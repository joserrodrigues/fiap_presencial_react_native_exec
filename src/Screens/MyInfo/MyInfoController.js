import React, { useState, useEffect} from 'react';
import MyInfoView from './MyInfoView';
import * as ImagePicker from 'expo-image-picker';
import * as Yup from "yup";
import { useDispatch } from 'react-redux';
import { changeName } from '../../store/modules/login/actions';
import { translate } from '../../Locales/ManageLocales';

const MyInfoController = ({ navigation }) => {
    const [image, setImage] = useState(null);
    const [showPhotoOptions, setShowPhotoOptions] = useState(false);
    const dispatch = useDispatch();

    //State que roda ao montar o componente
    const checkPermissions = async () => {

        //Checa Permissão para acessar a biblioteca de fotos
        let infoLibrary = await ImagePicker.requestMediaLibraryPermissionsAsync();
        if (infoLibrary.status !== 'granted') {
            alert(translate('alertPermissionGallery'));
            return false;
        }

        //Checa Permissão para acessar a Câmera
        let infoCamera = await ImagePicker.requestCameraPermissionsAsync();
        if (infoCamera.status !== 'granted') {
            alert(translate('alertPermissionCamera'));
            return false;
        }

        return true;
    }

    //Função para controlar a busca da image,
    const pickImage = async (type) => {

        let resultPermission = await checkPermissions();
        if(!resultPermission){
            return;
        }
        //Determina os parametros da imagem como tipo de imagem permitido, 
        //se permite editar a image, o formato da imagem e a qualidade
        let objectParams = {
            mediaTypes: ImagePicker.MediaTypeOptions.All,
            allowsEditing: true,
            aspect: [4, 3],
            quality: 1,
        }

        let result = null;
        if (type === 1) {
            //Busca as imagens da galeria
            result = await ImagePicker.launchImageLibraryAsync(objectParams);
        } else {
            //Busca a imagem da camera
            result = await ImagePicker.launchCameraAsync(objectParams);
        }
        if (!result.cancelled) {
            setImage(result.uri);
        }

        setShowPhotoOptions(false);
    };

    const toggleCameraOptions = () => {
        console.log(showPhotoOptions);
        setShowPhotoOptions(!showPhotoOptions);
    }

    const signInSchema = Yup.object().shape({
        name: Yup.string().required(translate('requiredName')),
        email: Yup.string().email(translate('invalidEmail')).required(translate('requiredEmail')),

        password: Yup.string()
            .required(translate('requiredPassword'))
            .min(4, translate('shortPassword')),
        confirmPassword: Yup.string()
            .required(translate('requiredConfirmPassword'))
            .oneOf([Yup.ref('password'), null], translate('passwordMustMatch'))                       
    });    

    const onSubmit = (values) => {
        console.log(values);
        dispatch(changeName(values.name))
    }
    return (
        <MyInfoView 
            navigation={navigation} 
            image={image}
            showPhotoOptions={showPhotoOptions}
            signInSchema={signInSchema}
            onSubmit={onSubmit}
            pickImage={pickImage}
            toggleCameraOptions={toggleCameraOptions}
         />
    );
};

export default MyInfoController;