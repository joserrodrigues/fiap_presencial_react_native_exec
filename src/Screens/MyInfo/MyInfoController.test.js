import React from 'react';
import renderer from 'react-test-renderer';
import MyInfoController from './MyInfoController';
import store from '../../store/stores';
import { Provider } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

jest.setTimeout(30000);
jest.useFakeTimers();

beforeAll(() => {

});

afterEach(() => {

});

describe('<MyInfoController />', () => {

    it('Check Initial Case', async () => {
        const navigation = useNavigation();
        const tree = renderer.create(
            <Provider store={store}>
                <MyInfoController navigation={navigation} />
            </Provider>
        ).toJSON();
        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot();
    });

    it('Check Empty Form', async () => {

        const navigation = useNavigation();
        const { getByText, getByTestId, getByPlaceholderText } = render(
            <Provider store={store}>
                <MyInfoController navigation={navigation} />
            </Provider>
        );

        const nameInput = getByPlaceholderText("Nome");
        expect(nameInput).toBeTruthy();

        const emailInput = getByPlaceholderText("E-mail");
        expect(emailInput).toBeTruthy();

        const passInput = getByPlaceholderText("Senha");
        expect(passInput).toBeTruthy();

        const confirmPassInput = getByPlaceholderText("Confirmar senha");
        expect(confirmPassInput).toBeTruthy();

        const button = getByTestId('saveButton');
        expect(button).toBeTruthy();
        fireEvent.press(button);

        let infoEmail = await waitFor(() => getByText('E-mail é obrigatório'));
        expect(infoEmail).toBeTruthy();
        let infoPass = await waitFor(() => getByText('Senha é obrigatório'));
        expect(infoPass).toBeTruthy();
    });

    it('Check Invalid E-mail', async () => {
        const navigation = useNavigation();
        const { getByText, getByTestId, getByPlaceholderText } = render(
            <Provider store={store}>
                <MyInfoController navigation={navigation}/>
            </Provider>
        );

        const emailInput = getByPlaceholderText("E-mail");
        expect(emailInput).toBeTruthy();

        const passInput = getByPlaceholderText("Senha");
        expect(passInput).toBeTruthy();

        const button = getByTestId('saveButton');
        expect(button).toBeTruthy();

        fireEvent.changeText(emailInput, 'hello');
        fireEvent.changeText(passInput, '123');
        fireEvent.press(button);

        await waitFor(() => {
            expect(getByText('E-mail não válido')).toBeTruthy();
            expect(getByText('Senha é curta - deveria ter ao menos 4 caracteres')).toBeTruthy();
        })
    });

    it('Check PassWord Must Match', async () => {
        const navigation = useNavigation();
        const { getByText, getByTestId, getByPlaceholderText } = render(
            <Provider store={store}>
                <MyInfoController navigation={navigation}/>
            </Provider>
        );

        const emailInput = getByPlaceholderText("E-mail");
        expect(emailInput).toBeTruthy();

        const passInput = getByPlaceholderText("Senha");
        expect(passInput).toBeTruthy();

        const confirmPassInput = getByPlaceholderText("Confirmar senha");
        expect(confirmPassInput).toBeTruthy();

        const button = getByTestId('saveButton');
        expect(button).toBeTruthy();

        fireEvent.changeText(emailInput, 'a@a.com');
        fireEvent.changeText(passInput, '1234');
        fireEvent.changeText(confirmPassInput, '123');
        fireEvent.press(button);

        await waitFor(() => {
            expect(getByText('As senhas devem ser iguais')).toBeTruthy();
        })
    });
});