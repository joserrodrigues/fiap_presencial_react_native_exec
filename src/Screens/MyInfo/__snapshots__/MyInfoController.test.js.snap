// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyInfoController /> Check Initial Case 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": 200,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": undefined,
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "width": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#242424",
                  "fontSize": 12,
                  "marginTop": 10,
                }
              }
            >
              Alterar Foto
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 2,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#86939e",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Nome
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#86939e",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginVertical": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    id={1}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Nome"
                    placeholderTextColor="#86939e"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#86939e",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  E-mail
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#86939e",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginVertical": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    id={1}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="E-mail"
                    placeholderTextColor="#86939e"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#86939e",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Senha
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#86939e",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginVertical": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    id={1}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Senha"
                    placeholderTextColor="#86939e"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#86939e",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Confirmar senha
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#86939e",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginVertical": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    id={1}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Confirmar senha"
                    placeholderTextColor="#86939e"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              />
            </View>
          </View>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <RNCSafeAreaView
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.2)",
                  "flex": 1,
                  "flexDirection": "column-reverse",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <View>
                <RCTScrollView>
                  <View>
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "borderColor": "#bcbbc1",
                            "flexDirection": "row",
                            "padding": 14,
                          }
                        }
                        testID="padView"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "flex-start",
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                          theme={
                            Object {
                              "colors": Object {
                                "black": "#242424",
                                "disabled": "hsl(208, 8%, 90%)",
                                "divider": "#bcbbc1",
                                "error": "#ff190c",
                                "grey0": "#393e42",
                                "grey1": "#43484d",
                                "grey2": "#5e6977",
                                "grey3": "#86939e",
                                "grey4": "#bdc6cf",
                                "grey5": "#e1e8ee",
                                "greyOutline": "#bbb",
                                "platform": Object {
                                  "android": Object {
                                    "error": "#f44336",
                                    "grey": "rgba(0, 0, 0, 0.54)",
                                    "primary": "#2196f3",
                                    "searchBg": "#dcdce1",
                                    "secondary": "#9C27B0",
                                    "success": "#4caf50",
                                    "warning": "#ffeb3b",
                                  },
                                  "default": Object {
                                    "error": "#ff3b30",
                                    "grey": "#7d7d7d",
                                    "primary": "#007aff",
                                    "searchBg": "#dcdce1",
                                    "secondary": "#5856d6",
                                    "success": "#4cd964",
                                    "warning": "#ffcc00",
                                  },
                                  "ios": Object {
                                    "error": "#ff3b30",
                                    "grey": "#7d7d7d",
                                    "primary": "#007aff",
                                    "searchBg": "#dcdce1",
                                    "secondary": "#5856d6",
                                    "success": "#4cd964",
                                    "warning": "#ffcc00",
                                  },
                                  "web": Object {
                                    "error": "#ff190c",
                                    "grey": "#393e42",
                                    "primary": "#2089dc",
                                    "searchBg": "#303337",
                                    "secondary": "#ca71eb",
                                    "success": "#52c41a",
                                    "warning": "#faad14",
                                  },
                                },
                                "primary": "#2089dc",
                                "searchBg": "#303337",
                                "secondary": "#ca71eb",
                                "success": "#52c41a",
                                "warning": "#faad14",
                                "white": "#ffffff",
                              },
                            }
                          }
                        >
                          <Text
                            onPress={[Function]}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#242424",
                                "fontSize": 17,
                                "margin": 15,
                              }
                            }
                            testID="listItemTitle"
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "fontSize": 15,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "marginRight": 10,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                              <Text
                                style={
                                  Object {
                                    "color": "#242424",
                                  }
                                }
                              >
                                Biblioteca
                              </Text>
                            </View>
                          </Text>
                          <Text
                            onPress={[Function]}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#242424",
                                "fontSize": 17,
                                "margin": 15,
                              }
                            }
                            testID="listItemTitle"
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "fontSize": 15,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "marginRight": 10,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                              <Text
                                style={
                                  Object {
                                    "color": "#242424",
                                  }
                                }
                              >
                                Câmera
                              </Text>
                            </View>
                          </Text>
                          <Text
                            onPress={[Function]}
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#242424",
                                "fontSize": 15,
                                "margin": 15,
                              }
                            }
                            testID="listItemTitle"
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "fontSize": 15,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "marginRight": 10,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                              <Text
                                style={
                                  Object {
                                    "color": "#fa5757",
                                  }
                                }
                              >
                                Cancelar
                              </Text>
                            </View>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </RNCSafeAreaView>
          </Modal>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 30,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "position": "absolute",
        "right": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            Object {
              "borderRadius": 28,
              "elevation": 4,
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="saveButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fa5757",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 56,
                "justifyContent": "center",
                "padding": 16,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 56,
                  "marginHorizontal": 5,
                  "overflow": "hidden",
                  "padding": 16,
                  "width": 56,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
